protocols/discord.py

```diff
@@ -487,7 +487,11 @@ def get_friendly_name(self, *args, **kwargs):
     def message(self, source, target, text, notice=False):
         """Sends messages to the target."""
         if target in self.users:
-            discord_target = self.users[target].discord_user.user.open_dm()
+            try:
+                discord_target = self.users[target].dm_channel
+            except ValueError:
+                discord_target = self.users[target].dm_channel = self.users[target].discord_user.user.open_dm()
```

Comment:
Unknown attributes raise AttributeError and not ValueError for me.

```
<GL> eval irc.users[source].test
-PyLinkDSC- Uncaught exception in command 'eval': AttributeError: 'User' object has no attribute 'test'
```

But either way, I think a better solution might be something like:

```python
userobj = self.users[target]
discord_target = getattr(userobj, 'dm_channel', userobj.discord_user.user.open_dm())
```

Which will automatically create a new channel if it cannot be found.

Reply:
Good call, it'd need to be something more like
```python
 userobj = self.users[target]
 discord_target = self.users[target].dm_channel = getattr(userobj, 'dm_channel', userobj.discord_user.user.open_dm())
 ```
To set the cached value